src/content/en/fundamentals/native-hardware/recording-video/index.md

```diff
@@ -0,0 +1,212 @@
+upload a file from the file system (ignoring `capture="camera"`). In Safari
+on iOS it will open up the camera app, allowing you to record video and
+then send it back to the web page; on Android it will give the user the
+choice of which app to use record the video in before sending it back to the web
+page.
+
+Once the user has finished recording and they are back in the website, you
+need to somehow get ahold of the file data. You can get quick access by
+attaching an `onchange` event to the input element and then reading
+the `files` property of the event object.
+
+<pre class="prettyprint">
+&lt;input type="file" accept="video/*" capture="camera" id="recorder">
+&lt;video id="player" controls>&lt;/video>
+&lt;script>
+  var recorder = document.getElementById('recorder');
+  var player = document.getElementById('player')'
+
+  recorder.addEventListener('change', function(e) {
+    var file = e.target.files[0];
+    // Do something with the video file.
+    player.src = URL.createObjectURL(file);
+  });
+&lt;/script>
+</pre>
+
+Once you have access to the file you can do anything you want with it. For
+example, you can:
+
+* Attach it directly to a `<video>` element so that you can play it
+* Download it to the user's device
+* Upload it to a server by attaching to an `XMLHttpRequest`
+* Draw the frames into a canvas and apply filters to it
+
+Whilst using the input element method of getting access to video data is
+ubiquitous, it is the least appealing option. We really want to get access to
+the camera and provide a nice experience directly in the page.
+
+## Access the camera interactively
+
+Modern browsers can have a direct line to the camera allowing us to build
+experiences that are fully integrated with the web page and the user will never
+leave the browser.
+
+### Acquire access to the camera
+
+We can directly access the camera by using an API in the WebRTC
+specification called `getUserMedia()`. `getUserMedia()` will prompt the user for
+access to their connected microphones and cameras.
+
+If successful the API will return a `Stream` that will contain the data from
+either the camera or the microphone, and we can then either attach it to
+a `<video>` element or save it using the `MediaRecorder` API.
+
+To get data from the camera we just set `video: true` in the constraints
+object that is passed to the `getUserMedia()` API
+
+<pre class="prettyprint">
+&lt;video id="player" controls>&lt;/video>
+&lt;script>
+  var player = document.getElementById('player');
+
+  var handleSuccess = function(stream) {
+    player.srcObject = stream;
+  };
+
+  navigator.mediaDevices.getUserMedia({ audio: true, video: true })
+      .then(handleSuccess)
+&lt;/script>
+</pre>
+
+By itself, this isn't that useful. All we can do is take the video data and play
+it back.
+
+### Access the raw data from the camera
+
+To access the raw video data from the camera you can draw each frame into a `<canvas>` and
```

Comment:
please show how to render it into a canvas. the live filter effects article is reasonable to link to but we need to show people how to take a frame and get it to the canvas first.

Reply:
The corollary is the audio article has web audio processing in as an example.